File: Problem One.js
const problemOne = max => {
  let sum = 0
  sum += sumMultiples(max, 3)
  sum += sumMultiples(max, 5)
  sum -= sumMultiples(max, 15)
  return sum
}

const sumMultiples = (max, multiple) => {
  max = Math.ceil(max/multiple - 1)
  let sum = 0, start = 0
  while(start <= max) {
    sum += start
    start++
  }
  return multiple * sum
}

const result = problemOne(1000)
console.log(result)
// 0.147s

File: problemOne.py
# If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

# Find the sum of all the multiples of 3 or 5 below 1000.

def main(max):
    result = sumMultiples(max, 3)
    result += sumMultiples(max, 5)
    result -= sumMultiples(max, 15)
    return result

def sumMultiples(max, multiple):
    if max % multiple == 0:
        max = (max / multiple) -1
    else:
        max /= multiple
    sum = 0
    start = 0
    while start <= max:
        sum += start
        start += 1
    return multiple * sum

print main(1000)
# 233168
# [Finished in 0.028s]
